pkg/controllers/sync/version/manager.go

```diff
@@ -70,38 +63,52 @@ type VersionManager struct {
 
 	versions map[string]runtimeclient.Object
```

Comment:
Would like to avoid changes to the PropagatedVersion mess as much as possible since we're uncertain what to do with it. If we decide to keep it, we could refactor it in the future for consistency.